### ISSUE TITLE
Nextcloud Desktop Hangs (on file upload) after update Linux[Bug]: 

### ISSUE THREAD

**Issue by gessel** [nextcloud/desktop#5267]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

The share dialog is looking incomplete.
...
The latest version of the nextcloud client (3.6.4-20221208.195402.3ffb9827b-1.0~jammy1) in my config hangs when uploading a file change with 🗹 automatic bandwidth limits set.  Settings→Network→Download Bandwidth→ 🔘 No Limit and  Upload Bandwidth→ 🔘 No Limit resolves the issue.

Bandwidth limits were set a long time ago and have worked normally through many iterations, but crashed.  The settings differences between a working desktop client and the failed one indicated that was the only difference between the two and unsetting the feature resolved the issue.

It may be related to #5242

[debug_client_FILES.zip](https://github.com/nextcloud/desktop/files/10213246/debug_client_FILES.zip)

### Steps to reproduce

1. 
Right click on the file
2. Click on Share Options
3. The share dialog does not display the option to share by e-mail
...
1. Leave bandwidth limits enabled (worked with previous version)
2. add file to sync folder
3. start the client
4. wait until it hangs using 100% CPU for as long as you are willing
5. force quit client  
6. CPU goes back to zero, file is not synced
7. No log entries on server
8. Client log stops on new file

### Expected behavior

When clicking on the share dialog, share by e-mail should be an option.
...
Files would sync as usual

### Which files are affected by this bug

recommended-install.txt simple_world.svg

### Operating system

Linux

### Which version of the operating system you are running.

5.15.0-56-generic "Linux Mint 21 Vanessa"

### Package

Distro package manager

### Nextcloud Server version

25.0.2

### Nextcloud Desktop Client version

3.6.4 (3.6.4-20221208.195402.3ffb9827b-1.0~jammy1)

### Is this bug present after an update or on a fresh install?

Updated from a minor version (ex. 3.4.2 to 3.4.4)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
No logs were generated (no problems reported) during the failed attempts.
```

### Additional info

Another computer connecting to the same server running the same client version can sync without error.  The problem client will download changes from the server but not sync changes up to the server.   The only significant difference is the bandwidth limits option and unsetting that seems to resolve the issue.

Info for working client:
<pre>
[General]
clientVersion=3.6.4-20221208.195402.3ffb9827b-1.0~jammy1
confirmExternalStorage=true
newBigFolderSizeLimit=500
optionalServerNotifications=true
showCallNotifications=true
useNewBigFolderSizeLimit=true

(obviously incredibly sensitive data which you really shouldn't be collecting and asking people to post redacted!)

[Settings]
geometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\x4\x1f\0\0\x1\xbc\0\0\tE\0\0\x6\x45\0\0\x4\x1f\0\0\x1\xeb\0\0\tE\0\0\x6\x45\0\0\0\0\0\0\0\0\xf\0\0\0\x4\x1f\0\0\x1\xeb\0\0\tE\0\0\x6\x45)

[SharingDialog]
geometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\x6\xb5\0\0\x3m\0\0\b7\0\0\x4\x96\0\0\x6\xb5\0\0\x3\x9c\0\0\b7\0\0\x4\x96\0\0\0\0\0\0\0\0\xf\0\0\0\x6\xb5\0\0\x3\x9c\0\0\b7\0\0\x4\x96)
</pre>

General info for failing client:
<pre>
[General]
clientVersion=3.6.4-20221208.195402.3ffb9827b-1.0~jammy1
confirmExternalStorage=true
crashReporter=true
logToTemporaryLogDir=true
monoIcons=false
newBigFolderSizeLimit=1508
optionalDesktopNotifications=true
optionalServerNotifications=true
showCallNotifications=true
useNewBigFolderSizeLimit=true

[BWLimit]
downloadLimit=80
uploadLimit=10
useDownloadLimit=-1
useUploadLimit=-1

[LogBrowser]
geometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\x6J\0\0\x3-\0\0\v{\0\0\x6\x1e\0\0\x6J\0\0\x3I\0\0\v{\0\0\x6\x1e\0\0\0\0\0\0\0\0\xf\0\0\0\x6J\0\0\x3I\0\0\v{\0\0\x6\x1e)

[Settings]
geometry=@ByteArray(\x1\xd9\xd0\xcb\0\x3\0\0\0\0\x5\xa1\0\0\0\xa4\0\0\v\x1d\0\0\a\xd1\0\0\x5\xa1\0\0\0\xc0\0\0\v\x1d\0\0\a\xd1\0\0\0\0\0\0\0\0\xf\0\0\0\x5\xa1\0\0\0\xc0\0\0\v\x1d\0\0\a\xd1)

[SharingDialog]
geometry=@ByteArray(\x1\xd9\xd0\xcb\0\x2\0\0\0\0\a'\0\0\x3S\0\0\b\xe1\0\0\x4\x9a\0\0\a*\0\0\x3l\0\0\b\xde\0\0\x4\x97\0\0\0\0\0\0\0\0\x10\0)
<pre>

**Comment by chrisjrob** [comment]
I have had permanent 100% CPU since yesterday on Ubuntu 22.04 with nextcloud-desktop 3.4.2. Tried upgrading to 3.6.4 AppImage - which was exactly the same. Found this post, and removed my BWLimits:

`[BWLimit]
downloadLimit=2048
uploadLimit=1024
useDownloadLimit=1
useUploadLimit=1
`

And restarted. The client CPU started at 25% and after a few seconds dropped off `top` entirely.

**Comment by fsegatoz** [comment]
I've the same issue with 3.4.6 client for Windows 10 with nextcloud snap 25.0.2, see https://help.nextcloud.com/t/desktop-client-no-longer-updates-files-and-hangs-windows-explorer/153167

**Comment by waveywhite** [comment]
This has affected me too. Removing bandwidth limitations solved issues with Nextcloud desktop hanging with 100% usage of a CPU core.
version:
nextcloud-desktop/kinetic,now 3.6.6-20230120.115834.ff2293824-1.0~kinetic1 amd64

**Comment by waltercool** [comment]
Can confirm this. Very annoying bug since there is no way to debug

**Comment by sebthemagnificient** [comment]
Same behaviour for me, removing bandwidth limitations also "solved" the problem.
Download works properly while upload does not ( can create empty files but cannot feed them nor upload anything else )

Nexcloud client 3.6.6, windows 10 22H2

**Comment by waltercool** [comment]
Any way to workaround this? Except removing the limits?

I do need that functionality.

**Comment by chrisjrob** [comment]
> Any way to workaround this? Except removing the limits?

If your nextcloud server is hosted behind an asynchronous connection, you may find as I did that it works just fine without the bandwidth limits, as the server's upload limit restricts the client's download limit.

Otherwise you could limit your server's upload bandwidth temporarily, there are various tools to do this, and a quick search suggested 'wondershaper'. This would obviously affect all your clients, so may not work for you.

**Comment by guidot** [comment]
I was affacted even though I did not set any limits. Removing the entire [BWLimit] section helped.

**Comment by merwan** [comment]
Same issue here after upgrading from Debian 11 to Debian 12 which upgraded Nextcloud desktop from 3.1.1 to 3.7.3.
Removing entirely the `[BWLimit]` section from config file `$HOME/.config/Nextcloud/nextcloud.cfg` solved the issue.

For information, the bandwidth limitation section was previously set to:
```ini
[BWLimit]                                                                                                                                                                                                          
downloadLimit=80                                                                                                                                                                                                   
uploadLimit=10                                                                                                                                                                                                     
useDownloadLimit=-1                                                                                                                                                                                                
useUploadLimit=-1 
```

**Comment by claucambra** [comment]
Hi folks, this is now fixed with #5680
